examples/AzureSDKDemoSwift/Source/BlobDownloadViewController.swift

```diff
@@ -79,25 +78,20 @@ class BlobDownloadViewController: UIViewController, MSALInteractiveDelegate {
                 self.tableView.reloadData()
             case let .failure(error):
                 self.showAlert(error: error)
-                self.noMoreData = true
             }
         }
     }
 
     /// Uses asynchronous "nextPage" method to fetch the next page of results and update the table view.
     private func loadMoreSettings() {
-        guard noMoreData == false else { return }
         dataSource?.nextPage { result in
             switch result {
             case let .success(data):
                 if data != nil {
```

Comment:
If I read the commentary right, this condition should now always be true right?

Reply:
If that's accurate, should we change the Result type for nextPage / nextItem to have a non-nullable element in the success case?

Reply:
Good point. I've updated those return types and adjusted accordingly. Now the client code doesn't need to worry about the "got nothing" scenario at all!

Reply:
Awesome, yeah that definitely simplifies things. Probably just need to update the demo app here to match that and then this is 👍 to merge.